src/services/item/service.ts

```diff
@@ -108,103 +112,276 @@ export class ItemService {
+    member: Member,
+    repositories: Repositories,
+    items: (Partial<Item> & Pick<Item, 'name' | 'type'>)[],
+    parentId?: string,
+    previousItemId?: string,
+  ) {
+    // name and type should exist
+    for (const item of items) {
+      if (!item.name || !item.type) {
+        throw new MissingNameOrTypeForItemError(item);
       }
     }
 
-    this.log.debug(`create item ${item.name}`);
-    const createdItem = await itemRepository.addOne({
-      item,
-      creator: member,
+    let createdItems: Item[];
+    if (parentId) {
+      createdItems = await this.createItemsWithParent(
+        member,
+        repositories,
+        items,
+        parentId,
+        previousItemId,
+      );
+    } else {
+      createdItems = await this.createItemsAndMemberships(member, repositories, items, null);
+    }
+
+    // index the items for search
+    this.indexItemsForSearch(createdItems, repositories);
+
+    return createdItems;
+  }
+
+  private async createItemsWithParent(
+    member: Member,
+    repositories: Repositories,
+    items: (Partial<Item> & Pick<Item, 'name' | 'type'>)[],
+    parentId: string,
+    previousItemId?: string,
+  ) {
+    const { itemRepository, itemMembershipRepository } = repositories;
+
+    this.log.debug(`verify parent ${parentId} exists and the member has permission over it`);
+    const parentItem = await this.get(member, repositories, parentId, PermissionLevel.Write);
+    const inheritedMembership = await itemMembershipRepository.getInherited(
+      parentItem.path,
+      member.id,
+      true,
+    );
+
+    // quick check, necessary for ts
+    if (!isItemType(parentItem, ItemType.FOLDER)) {
+      throw new ItemNotFolder(parentItem);
+    }
+
+    itemRepository.checkHierarchyDepth(parentItem);
+
+    // check if there's too many children under the same parent
+    const descendants = await itemRepository.getChildren(member, parentItem);
+    if (descendants.length + items.length > MAX_NUMBER_OF_CHILDREN) {
+      throw new TooManyChildren();
+    }
+
+    // no previous item adds at the beginning
+    // else define order from given previous item id
+    let order;
+    if (!previousItemId) {
+      order = await repositories.itemRepository.getFirstOrderValue(parentItem.path);
+    } else {
+      order = await repositories.itemRepository.getNextOrderCount(parentItem.path, previousItemId);
+    }
+    for (let i = 0; i < items.length; i++) {
+      items[i] = { ...items[i], order: order++ };
+    }
+
+    const createdItems = await this.createItemsAndMemberships(
+      member,
+      repositories,
+      items,
+      inheritedMembership,
       parentItem,
-    });
-    this.log.debug(`item ${item.name} is created: ${createdItem}`);
+    );
+
+    // rescale the item ordering, if there's more than one item
+    if (items.length > 1) {
+      repositories.itemRepository.rescaleOrder(member, parentItem);
+    }
+
+    return createdItems;
+  }
+
+  private async createItemsAndMemberships(
```

Comment:
Add comments about this function and the other one, how it is different from the one above